4p3 Documentation (#66)
ucgmsim/nzgmdb: wiki/Process-Records.md

  - `fmin_v`: `fmin_mean` value from vertical component (Z)

**2. Fmax Extraction**
- Loads Fmax values from `fmax.csv` 
- **Component-specific fmax extraction**:
  - `fmax_h`: Minimum Fmax from horizontal components (000, 090)
  - `fmax_v`: Fmax value from vertical component (ver)
- Ensures frequency band validity for processing

**3. Bypass Records Support**
- Optionally loads custom frequency bounds from bypass records file
- Overrides GMC/Fmax values with user-specified component-specific fmin/fmax when provided
- Handles NaN values appropriately during override process

### 🔹 Waveform Processing Pipeline

Each MSEED file undergoes a comprehensive processing workflow:

#### Initial Preprocessing
1. **Component Validation** - Ensures all 3 components (000, 090, ver) are present
2. **Demean and Detrend** - Remove offset and linear trends
3. **Taper Application** - Apply 5% Tukey taper to both ends
4. **Zero Padding** - Add 5 seconds of zeros at start and end
5. **Inventory Response Removal** - Remove instrument sensitivity using station metadata
6. **Component Rotation** - Rotate horizontal components to North-East-Vertical (NEZ)

> 7. **Gravity normalisation** - Divide acceleration data by gravitational constant (9.81 m/s²)

Reviewer: The gravitational constant is actually a different quantity
```suggestion
7. **Gravity normalisation** - Divide acceleration data by the acceleration due to gravity (9.81 m/s²)
```

- **Component Consistency** - Process all components with their respective optimal parameters

### 🔹 Configuration Parameters

Key configuration values from `config.yaml`:

| Parameter | Default Value | Description |
|-----------|---------------|-------------|
| `taper_fraction` | 0.05 | Fraction for Tukey taper application |
| `zero_padding_time` | 5 | Zero padding duration (seconds) |
| `g` | 9.81 | Gravitational acceleration (m/s²) |
| `order_default` | 4 | Butterworth filter order |
| `poly_order_default` | 6 | Polynomial detrending order |
| `low_cut_default` | 0.04 | Default low-cut frequency if fmin unavailable |

---

## 📦 Output

### 🔹 Processed Waveform Files

The primary output consists of ASCII text files stored in the directory structure:
```
waveforms/
├── year/
│   └── event_id/
│       └── processed/
│           ├── evid_station_channel_location.000  # North component
│           ├── evid_station_channel_location.090  # East component  
│           └── evid_station_channel_location.ver  # Vertical component
```

Each ASCII file contains:
- **Header Line**: Metadata including station, channel, sampling rate, number of points

> - **Data Lines**: One acceleration value per line in m/s² units

Reviewer: units in g's have been mentioned before, so I just wanted to check that this is correct being m/s² (and not g)
